core/include/detray/propagator/rk_stepper.ipp

```diff
@@ -142,7 +186,9 @@ auto detray::rk_stepper<magnetic_field_t, transform3_t, constraint_t, policy_t,
     -> vector3 {
     auto& track = this->_track;
 
-    const auto qop = track.qop();
+    // const auto qop = track.qop();
+
+    const auto qop = evaluate_qop(h);
```

Comment:
Can you calculate the value for ```qop``` outside of this function (e.g. in ```try_rk4```), so that ```evaluate_qop``` is called only once per new step size, instead of four times?

Reply:
I fixed the code to calculate two times -> one at step_size/2 and the other at step_size.

Reply:
Is the code still like that? I thought i removed it

Reply:
I removed some caching variables. There are only k_qop4 now. Is this good?

Reply:
I don't understand why you want to take out qop from `evaulate_k`. That's what code was like before.
I believe evaluate_k should include qop inside it because that's what `k` is defined for.

Reply:
> Is the code still like that? I thought i removed it

This is something that I wrote in mistake definitely. I don't want to get rid of qop from `evaluate_k` function

Reply:
I guess the compiler inlines it, then it won't make a difference